### ISSUE TITLE
Compare with RecombinHunt

### ISSUE THREAD

**Issue by jeromekelleher** [jeromekelleher/sc2ts-paper#195]
https://www.biorxiv.org/content/10.1101/2023.06.05.543733v1

**Comment by szhan** [comment]
It's published in now.

Alfonsi, T., Bernasconi, A., Chiara, M. et al. Data-driven recombination detection in viral genomes. Nat Commun 15, 3313 (2024). https://doi.org/10.1038/s41467-024-47464-5

**Comment by szhan** [comment]
The RecombinHunt authors consider the Pango X lineages as the ground truth. When they compare their method to RIPPLES/RIVET, they focus on how well they picked up the Pango X lineages as recombinants versus RIPPLES/RIVET (see Table 5).

**Comment by hyanwong** [comment]
Do they find many cases where a single recombination event (plus subsequent mutation) explains multiple PangoX lineages (like we find for XE/XH and XQ/XR/XAA/XAM/XAG)?

**Comment by szhan** [comment]
> Do they find many cases where a single recombination event (plus subsequent mutation) explains multiple PangoX lineages (like we find for XE/XH and XQ/XR/XAA/XAM/XAG)?

No, I don't see any comment about this. One would need to put the lineages in some phylogenetic context to figure it out, right? RecombinHunt doesn't do any phylogenetic inference.

**Comment by szhan** [comment]
This is one key claim the RecombinHunt authors make: "RecombinHunt correctly flags **33** lineages designated as recombinant by Pango, which however are **not identified as recombinant by RIPPLES/RIVET**, hence demonstrating a more than **2fold increase in sensitivity** with respect to the current state-of-the-art method for the detection of viral recombination at pandemic/epidemic scale."

**Comment by szhan** [comment]
The RecombinHunt authors have done some curating of the info for the Pango X lineages. We can take the info in their Table 2 (https://www.nature.com/articles/s41467-024-47464-5/tables/2).

**Comment by szhan** [comment]
There is a _ton_ of useful curated info about the Pango X lineages in this HTML file (including the ground-truth Pango parents and breakpoint intervals): SupplementaryNotes2_RH_gisaid_sarscov2_detail.html

It's a bit inconvenient to access though, because you have to download the zipped file first.
https://zenodo.org/records/13349272

**Comment by szhan** [comment]
For example, for this Pango X lineage, there are these info.

![Image](https://github.com/user-attachments/assets/bef73118-7a37-49b4-87fe-bf31a881d948)

**Comment by szhan** [comment]
Note that the RecombinHunt results for XB, XBA, XBQ, and XBR are not available in Table 2 ("Summary of results on GISAID dataset") (not sure why). When assembling the methods comparison table, I had to dig those results out from `SupplementaryNotes2_RH_gisaid_sarscov2_detail.html`. I hope we don't have to automate this, because if we do, we have to parse this HTML file for the Pango labels of the parents and the breakpoint intervals...

**Comment by szhan** [comment]
Note that I had to consult the alias key when comparing the Pango parents with the following labels:
BJ in BA
CJ in BA
BM in BA
BF in BA
BN in BA
Q = B.1.1.7

**Comment by szhan** [comment]
Note that there are two recombinant cases where RecombinHunt-GISAID identifies as having two breakpoints but the ground truth from `pango-designation` indicates as having one breakpoint.

pango | left_pango_parent | right_pango_parent | breakpoint_interval
-- | -- | -- | --
XBH | BA.2.75.2 | BA.2.3.17;BA.2.75.2 | 3039-3040;15713-15714
XBM | BF.3 | BA.2.76;BF.3 | 4330-4331;22673-22674

**Comment by szhan** [comment]
RecombinHunt does "gap resolution" at the end when the breakpoint positions are not adjacent, so the breakpoint intervals are narrow. The left coordinate is the rightmost position assigned to the left parent, and the right coordinate is the leftmost position assigned to the right parent.

**Comment by szhan** [comment]
There are some cases where results are available for RecombinHunt-GISAID but not RecombinHunt-Nextstrain and vice versa and where results of the two differ.

For example, RecombinHunt-GISAID results for XBQ and XBR are not available, but RecombinHunt-Nextstrain results are available (both identified as recombinants).

Also, for example, XBF is identified as a non-recombinant in the RecombinHunt-GISAID results, but it is identified as a recombinant in the RecombinHunt-Nextstrain results.

**Comment by szhan** [comment]
For RecombinHunt, the results clearly can differ depending on the input dataset (GISAID versus Nextstrain). Not sure how to do the comparison here.

Take the GISAID results over the Nextstrain results?
Merge them somehow? (This seems less fair to sc2ts though)

**Comment by jeromekelleher** [comment]
How does this intersect with the "easy set" vs "messy stuff"? I'm sort of expecting that the things that are coming up as slightly odd for us will also have discordant across other methods.

**Comment by szhan** [comment]
Briefly reopening to adjust the breakpoint intervals.

The RecombinHunt breakpoint intervals are defined by the left and right informative sites. To make them right-exclusive, we have to add one to the right coordinate. I'm updating `bigtable_rhgisaid.csv` and `bigtable_rhnextstrain.csv` to store the adjusted breakpoints, so that there is no need to adjust them before comparing them to the sc2ts breakpoint intervals.
